hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -378,27 +384,31 @@ object EmitCode {
 
   def fromCodeTuple(pt: PType, ct: IndexedSeq[Code[_]]): EmitCode = {
     if (pt.required)
-      new EmitCode(Code._empty, const(false), pt.fromCodeTuple(ct))
+      EmitCode(Code._empty, const(false), pt.fromCodeTuple(ct))
     else
-      new EmitCode(Code._empty, coerce[Boolean](ct.last), pt.fromCodeTuple(ct.init))
+      EmitCode(Code._empty, coerce[Boolean](ct.last), pt.fromCodeTuple(ct.init))
   }
 }
 
-case class EmitCode(setup: Code[Unit], m: Code[Boolean], pv: PCode) {
+class EmitCode(private val start: CodeLabel, private val iec: IEmitCode) {
+  def pv: PCode = iec.value
+
+  def setup: Code[Unit] = Code._empty
```

Comment:
I suspect this may break tests. Do all users of EmitCode always check missingness?

Reply:
Ah, you're probably right. Hmm...